### DISCUSSION TITLE
Connecting to MS SQL Express instance

### DISCUSSION THREAD

**Discussion by simon-lewis** [Budibase/budibase#9935] [Help]
I am struggling to connect Budibase to an MS SQLExpress instance. It may be that we need the ability to enter an INSTANCE NAME as part of the connection string. In my case, this would be SERVERNAME\**SQLEXPRESS** and a non-default port, where SQLEXPRESS is the instance name allowed for a SQL Express installation. This is available as a config option for the Connection object in the tedious library (which I think BB uses, from other articles) but is not exposed in the connection options form in Budibase. 

**options.instancename** in the tedious library Connection(config)

Is it possible to surface the INSTANCENAME as an optional configuration entry for an MS SQL Connection ?

**Comment by ConorWebb96** [comment]
Hey @simon-lewis,

For the server property could you try something like the below:
`<computerName>\<InstanceName> `

Input this into the server area in the screenshot below. possibly add a double backslash in to avoid it being escaped.

![ec7639b-create-ds-sales](https://user-images.githubusercontent.com/126772285/223666142-8d45c168-8fbf-4f60-a98e-09f9801dba84.png)

Regex expression highlighting instanceName:
<img width="372" alt="Screenshot 2023-03-08 at 08 46 31" src="https://user-images.githubusercontent.com/126772285/223665840-d1fd7f23-2147-4b50-b0e7-a0b7d0fd71ad.png">

**Comment by simon-lewis** [comment]
Thanks Conor @ConorWebb96 but no joy. Budibase connections work well with other SQL Express instances on the standard port (1433) and where there is no Instance Name in the connection. I will have to look at the configuration of this specific server  as it uses a non-standard port (which I can specify in the Budibase connection info) - something there is not right. I'll close this one for now but really appreciate you jumping in to assist.

For info. Tried `<ipaddress>\SQLEXPRESS `and `<ipaddress>\\SQLEXPRESS.` The first onetimes out; the second (2 slashes, \\) comes back with a DNS error (`ConnectionError: Failed to lookup instance on <ipaddress>\ - getaddrinfo ENOTFOUND <ipaddress>\`). Also tried with 3 slashes; same result.

**Comment by simon-lewis** [comment]
Hey @ConorWebb96 - abject apologies as this turns out to have been a 'user issue' (i.e. mine !). 
It WAS a firewall issue for this particular instance. Sorry to have wasted your time !
